### PR TITLE
[IMP] web, mail, *: initialize use readonly route

### PR DESCRIPTION
**Pull request by phenix-factory**
* = im_livechat, mail_bot, test_discuss_full

Backport of https://github.com/odoo/odoo/pull/179352

OdooBot onboarding needs to be moved to webclient load.

Write access is not useful since settings were moved out of the init messaging.

See https://github.com/odoo/odoo/pull/150625

### CODE REVIEW

File: addons/mail_bot/models/discuss_channel.py [hunk 1]
```diff
@@ -10,3 +9,7 @@ class Channel(models.Model):
     def execute_command_help(self, **kwargs):
         super().execute_command_help(**kwargs)
         self.env['mail.bot']._apply_logic(self, kwargs, command="help")  # kwargs are not usefull but...
+
+    def _message_post_after_hook(self, message, msg_vals):
+        self.env["mail.bot"]._apply_logic(self, msg_vals)
```
Comment (seb-odoo) [inline addons/mail_bot/models/discuss_channel.py:14]: If you move it here, it should be removed from thread.

It looks like the 2 removed files were not properly backported.

File: addons/mail/static/src/core/common/thread_model.js [hunk 2]
```diff
@@ -197,11 +197,6 @@ export class Thread extends Record {
         }
         return this.message_needaction_counter;
     }
-    isCorrespondentOdooBot = Record.attr(undefined, {
```
Comment (seb-odoo) [inline addons/mail/static/src/core/common/thread_model.js:200]: I'd not go as far as removing field in stable.

File: addons/mail_bot/models/mail_bot.py [hunk 3]
```diff
@@ -289,16 +293,6 @@ def _body_contains_emoji(self, body):
...
 
-    def _is_bot_pinged(self, values):
-        odoobot_id = self.env['ir.model.data']._xmlid_to_res_id("base.partner_root")
-        return odoobot_id in values.get('partner_ids', [])
-
-    def _is_bot_in_private_channel(self, record):
-        odoobot_id = self.env['ir.model.data']._xmlid_to_res_id("base.partner_root")
-        if record._name == 'discuss.channel' and record.channel_type == 'chat':
-            return odoobot_id in record.with_context(active_test=False).channel_partner_ids.ids
-        return False
```
Comment (seb-odoo) [inline addons/mail_bot/models/mail_bot.py:300]: I would not remove these two methods in stable.

**Review by seb-odoo — COMMENTED** [review]
This is a bit borderline as change in stable but with few adaptations it might be in the ok range.

File: addons/mail_bot/models/discuss_channel.py (hunk 1 above)
Comment (phenix-factory) [reply addons/mail_bot/models/discuss_channel.py:14]: Bad picking on my part :)

File: addons/mail_bot/models/mail_bot.py [hunk 4]
```diff
@@ -12,26 +12,28 @@ class MailBot(models.AbstractModel):
     _name = 'mail.bot'
     _description = 'Mail Bot'
 
-    def _apply_logic(self, record, values, command=None):
+    def _apply_logic(self, channel, values, command=None):
```
Comment (seb-odoo) [inline addons/mail_bot/models/mail_bot.py:15]: I'd also avoid the rename here.

File: addons/mail_bot/models/mail_bot.py [hunk 5]
```diff
@@ -48,10 +50,12 @@ def _get_style_dict():
             "paperclip_icon": Markup("<i class='fa fa-paperclip' aria-hidden='true'/>"),
         }
 
-    def _get_answer(self, record, body, values, command=False):
+    def _get_answer(self, channel, body, values, command=False):
```
Comment (seb-odoo) [inline addons/mail_bot/models/mail_bot.py:53]: And I'd also avoid the rename here.

**Review by seb-odoo — APPROVED** [review]

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**